cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -1490,14 +1489,20 @@ private module Cached {
       call.getStaticCallTarget() = f and
       // nodeFrom is a post-update node whose pre-update node is an out node
       outNode = nodeFrom.(PostUpdateNode).getPreUpdateNode() and
-      outNode.getCallInstruction() = call and
-      outNode.getIndirectionIndex() = indirectionIndex and
+      outNodeHasCallAndIndirection(outNode, call, indirectionIndex) and
       // and nodeTo is the post-update node whose pre-update node is the qualifier of the function.
       hasOperandAndIndex(nodeTo.(PostUpdateNode).getPreUpdateNode(), call.getThisArgumentOperand(),
         indirectionIndex)
     )
   }
 
+  private predicate outNodeHasCallAndIndirection(
```

Comment:
You probably want to `pragma[nomagic]` this as well to prevent inlining from undoing this.